### PR TITLE
Add polyfill documentation

### PR DESCRIPTION
**Pull request by darraghjburke**
Adds some basic documentation pages for the polyfills. Limited info at the moment so please comment anything else you think should be added.

### CODE REVIEW

File: Documentation/Polyfills.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Babylon Native Polyfills
+Babylon Native Polyfills add support for certain Javascript browser APIs, such as [`canvas`](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API) and [`window`](https://developer.mozilla.org/en-US/docs/Web/API/console). They do not attempt to fulfill the web specification exactly, nor do they mirror the behavior of browsers. The polyfills only provide functionality that is either required by Babylon.js, or would be frequently used by Babylon developers.
```
Comment (bghgary) [inline Documentation/Polyfills.md:2]: nit: multiple places
```suggestion
Babylon Native Polyfills add support for certain JavaScript browser APIs, such as [`canvas`](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API) and [`window`](https://developer.mozilla.org/en-US/docs/Web/API/console). They do not attempt to fulfill the web specification exactly, nor do they mirror the behavior of browsers. The polyfills only provide functionality that is either required by Babylon.js, or would be frequently used by Babylon developers.
```

File: Polyfills/Window/readme.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Window
+Implements parts of the `window` object. Currently supports:
+* `setTimeout()` but not `clearTimeout()`, `setInterval()`, or `clearInterval()`
+* `addEventListener()` and `removeEventListener()`
```
Comment (bghgary) [inline Polyfills/Window/readme.md:4]: This is false. These functions are stubs.

File: Polyfills/XMLHttpRequest/readme.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+We do not support `onload`-style event listeners. Instead, you should listen to events using `addEventListener`. At the moment, we only support the following events:
+* `loadend`
+* `readystatechange`
+
+## Local files
+Unlike the web, XMLHttpRequest supports loading local files using two schemes:
+* `file:///` allows you to load from an absolute path
+* `app:///` allows you to load from a relative path, either the current program or package depending on platform
+
+
```
Comment (bghgary) [inline Polyfills/XMLHttpRequest/readme.md:14]: nit: lots of space for no reason?
```suggestion
```

File: Polyfills/Window/readme.md (hunk 2 above)
Comment (darraghjburke) [reply Polyfills/Window/readme.md]: good catch! thanks

**Pull request merged**